pkg/gadgets/trace/dns/tracer/gc.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tracer
+
+import (
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// Delay between each garbage collection run.
+const garbageCollectorInterval = 1 * time.Second
+
+// Max age of query before it will be garbage collected.
+const garbageCollectorMaxQueryAge = 10 * time.Second
```

Comment:
hmm, I don't have hard data here. I see that [`nslookup`](https://elixir.bootlin.com/busybox/latest/source/networking/nslookup.c#L331), [`dig`](https://linux.die.net/man/1/dig), and [coredns](https://github.com/coredns/coredns/blob/c1b479fd024c91d03608587765dc9ae8745aef0c/plugin/forward/forward.go#L259) all set a default timeout of 5 seconds, although a user could configure it higher.

Possibly we could make this user-configurable? Or set the default much higher (60 seconds?)

Reply:
Updated to make query max age configurable (I called it "dns-timeout" but could change the name to whatever seems clearest).

I think I'll defer sending an event since that's tracked in a separate issue https://github.com/inspektor-gadget/inspektor-gadget/issues/1512